js/packages/ufg-client/test/it/fetch-resource.spec.ts

```diff
@@ -4,22 +4,45 @@ import {makeTestServer, generateCertificate} from '@applitools/test-server'
 import assert from 'assert'
 
 describe('fetch-resource', () => {
-  let server: any
+  let server: any, authority: any
 
   before(async () => {
-    const authority = await generateCertificate({days: 1})
-    server = await makeTestServer({...authority, port: 12345})
+    authority = await generateCertificate({days: 1})
   })
 
-  after(async () => {
+  afterEach(async () => {
     await server.close()
   })
 
   it('works with a self-signed certificate', async () => {
+    server = await makeTestServer({...authority})
     const fetchResource = makeFetchResource({retryLimit: 0})
     const resource = await fetchResource({
       resource: makeResource({url: `https://localhost:${server.port}/page/smurfs.jpg`}),
     })
     assert.strictEqual((resource.hash as any).contentType, 'image/jpeg')
   })
+
+  it('does not hang for unresponsive resource', async () => {
+    server = await makeTestServer({
+      ...authority,
+      middlewares: [
+        async (_req: any, _res: any, next: any) => {
+          await new Promise(resolve => setTimeout(resolve, 1200))
+          next()
+        },
+      ],
```

Comment:
There is already a middleware that may fit your need. Will it work for you?
```suggestion
      middlewares: ['slow'],
```